orc8r/gateway/python/scripts/register.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/sudo /usr/bin/env python3
+
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import asyncio
+import os
+import subprocess
+import sys
+from typing import List
+
+import grpc
+import snowflake
+
+from orc8r.protos.identity_pb2 import AccessGatewayID, Identity
+from orc8r.protos.bootstrapper_pb2_grpc import BootstrapperStub
+from magma.common.cert_utils import load_public_key_to_base64der
+from orc8r.protos.bootstrapper_pb2 import (
+    RegisterRequest,
+    RegisterResponse,
+    GetGatewayDeviceInfoRequest,
+    GetGatewayDeviceInfoResponse,
+)
+from orc8r.protos.bootstrapper_pb2_grpc import (
+    RegistrationStub,  # Register
+    CloudRegistrationStub,  # Register
+)
+from magma.common.service_registry import ServiceRegistry
+
+
+def bare_grpc_wrapper(func):
+    """
+    Wraps a function with a gRPC wrapper which creates a RPC client to
+    the service and handles any RPC Exceptions.
+
+    Usage:
+    @cloud_grpc_wrapper
+    def func(client, args):
+        pass
+    func(args, ProtoStubClass, 'service')
+    """
+
+    def wrapper(*alist) -> object:
+        args = alist[0]
+        stub_cls = alist[1]
+        chan = ServiceRegistry.get_bare_rpc_channel()
+        client = stub_cls(chan)
+        try:
+            func(client, args)
+        except Exception as err:
+            print(err)
+            exit(1)
+
+    return wrapper
+
+
+@bare_grpc_wrapper
+def register_handler(client: RegistrationStub, args: List[str]) -> RegisterResponse:
+    """
+    Registers a device and retrieves its control proxy
+    
+    Returns:
+        RegisterResponse: response from gRPC call, either error or the control_proxy
+    """
+    req = RegisterRequest()
+    req.token = args.token
+    req.hwid.id = snowflake.snowflake()
+    req.challenge_key.key = load_public_key_to_base64der(
+        "/var/opt/magma/certs/gw_challenge.key")
+    req.challenge_key.key_type = 0
+
+    res = client.Register(req)
+    if res.HasField("error"):
+        raise Exception(res.error)
+
+    print("Successfully registered gateway")
+    print("Hardware ID\n-----------\n%s" % req.hwid)
+    print("Challenge Key\n-----------\n%s\n" % req.challenge_key)
+    print("Control Proxy\n-----------\n%s\n\n" % res.control_proxy)
+
+    return res
+
+
+@bare_grpc_wrapper
+def get_gateway_device_info_handler(client: CloudRegistrationStub, args: List[str]) -> GetGatewayDeviceInfoResponse:
```

Comment:
As noted, this isn't necessary

And then, because we are only making a single request to the cloud, we may not need to use the "bare_grpc_wrapper" that you created -- just make the request manually, e.g. https://github.com/magma/magma/blob/33cd34820810ed71947aca9a6011182496c71324/orc8r/gateway/python/scripts/checkin_cli.py#L175-L194